src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.ParameterViewModels.cs

```diff
@@ -0,0 +1,246 @@
+            public abstract bool IsRemoved { get; set; }
+            public abstract bool IsDisabled { get; }
+            public abstract string CallSite { get; }
+            public abstract bool IsRequired { get; }
+            public abstract string DefaultValue { get; }
+
+            public abstract string InitialIndex { get; }
+            public abstract string Modifier { get; }
+            public abstract string Default { get; }
+
+            public ParameterViewModel(ChangeSignatureDialogViewModel changeSignatureDialogViewModel)
+            {
+                ChangeSignatureDialogViewModel = changeSignatureDialogViewModel;
+            }
+
+            public Visibility HasParameterNameConflict { get; set; }
+
+            public virtual Visibility TypeWarningVisibility => Visibility.Collapsed;
+
+            public string ShortAutomationText => $"{Type} {ParameterName}";
+
+            public virtual string FullAutomationText
+            {
+                get
+                {
+                    var text = $"{Modifier} {Type} {Parameter}";
+                    if (!string.IsNullOrWhiteSpace(Default))
+                    {
+                        text += $" = {Default}";
+                    }
+
+                    return text;
+                }
+            }
+
+            public bool NeedsBottomBorder
+            {
+                get
+                {
+                    if (this == ChangeSignatureDialogViewModel._thisParameter)
+                    {
+                        return true;
+                    }
+
+                    if (this == ChangeSignatureDialogViewModel._parametersWithoutDefaultValues.LastOrDefault() &&
+                        (ChangeSignatureDialogViewModel._parametersWithDefaultValues.Any() || ChangeSignatureDialogViewModel._paramsParameter != null))
+                    {
+                        return true;
+                    }
+
+                    if (this == ChangeSignatureDialogViewModel._parametersWithDefaultValues.LastOrDefault() &&
+                        ChangeSignatureDialogViewModel._paramsParameter != null)
+                    {
+                        return true;
+                    }
+
+                    return false;
+                }
+            }
+
+            public virtual string CallSiteValue => string.Empty;
+        }
+
+        public class AddedParameterViewModel : ParameterViewModel
+        {
+            public override Parameter Parameter => _addedParameter;
+            private readonly AddedParameter _addedParameter;
+
+            public AddedParameterViewModel(ChangeSignatureDialogViewModel changeSignatureDialogViewModel, AddedParameter addedParameter)
+                : base(changeSignatureDialogViewModel)
+            {
+                _addedParameter = addedParameter;
+            }
+
+            /// <summary>
+            /// Display text for the type in the UI, including a marker if it doesn't bind.
```

Comment:
this would be good for `/// <inherit-doc /> `

Reply:
Actually don't need it at all anymore. It was only there for clarification about the "doesn't bind" "marker" which is no more.